features/step_definitions/receipt_steps.rb

```diff
@@ -0,0 +1,78 @@
+require 'byebug'
+require './lib/tax'
+require './lib/discount'
+require './lib/payment'
+
+World(KnowsTheDomain)
+
+FLOAT = Transform /^\d*\.\d*$/ do |float_string|
+  float_string.to_f
+end
+
+Given(/^I am at a cafe with a sales tax of "(.*?)"$/) do |tax|
+  receipt.evaluators[:tax] = Tax.new(tax.to_f)
```

Comment:
You're breaking encapsulation a lot in these tests here, this for example you be `receipt.add_tax(Tax.new(tax.to_f))` or something